File: PJS5_CSharp/PJS5_CSharp/Sources/GUI/GUI.cs
﻿using PJS5_CSharp.Sources;
using PJS5_CSharp.Sources.Pilot;
using PJS5_CSharp.Sources.Robot;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GUI
{
    public static class Gui
    {
        public static void ShowStatus(Robot pPlayer, Robot pBot)
        {
            Console.WriteLine("|====================--====================|");
            Console.WriteLine("|                  Robots                  |");
            Console.WriteLine("|===================-<>-===================|");
            Console.WriteLine("|>-     Player     -<||>-     Ennemi     -<|");
            Console.WriteLine("|===================-<>-===================|");
            Console.WriteLine("| Furnace armor: " + pPlayer.GetFurnaceArmor() + "/" + pPlayer.GetFurnaceMaxArmor() + " || Furnace armor: " + pBot.GetFurnaceArmor() + "/" + pBot.GetFurnaceMaxArmor() + " |");
            Console.WriteLine("| Furnace life: " + pPlayer.GetFurnaceLife() + "/" + pPlayer.GetFurnaceMaxLife() + "  || Furnace life: " + pBot.GetFurnaceLife() + "/" + pBot.GetFurnaceMaxLife() + "  |");
            Console.WriteLine("|>- - - - - - - - - -<>- - - - - - - - - -<|");
            Console.WriteLine("| Legs armor: " + pPlayer.GetLegsArmor() + "/" + pPlayer.GetLegsMaxArmor() + "    || Legs armor: " + pBot.GetLegsArmor() + "/" + pBot.GetLegsMaxArmor() + "    |");
            Console.WriteLine("| Legs remains: " + pPlayer.GetLegsLife() + "/" + pPlayer.GetLegsMaxLife() + "  || Legs remains: " + pBot.GetLegsLife() + "/" + pBot.GetLegsMaxLife() + "  |");
            Console.WriteLine("|>- - - - - - - - - -<>- - - - - - - - - -<|");
            Console.WriteLine("| WeaponL armor: " + pPlayer.GetLeftWeaponArmor() + "/" + pPlayer.GetLeftWeaponMaxArmor() + " || WeaponL armor: " + pBot.GetLeftWeaponArmor() + "/" + pBot.GetLeftWeaponMaxArmor() + " |");
            Console.WriteLine("| WeaponL life: " + pPlayer.GetLeftWeaponLife() + "/" + pPlayer.GetLeftWeaponMaxLife() + "  || WeaponL life: " + pBot.GetLeftWeaponLife() + "/" + pBot.GetLeftWeaponMaxLife() + "  |");
            Console.WriteLine("|>- - - - - - - - - -<>- - - - - - - - - -<|");
            Console.WriteLine("| WeaponR armor: " + pPlayer.GetRightWeaponArmor() + "/" + pPlayer.GetRightWeaponMaxArmor() + " || WeaponR armor: " + pBot.GetRightWeaponArmor() + "/" + pBot.GetRightWeaponMaxArmor() + " |");
            Console.WriteLine("| WeaponR life: " + pPlayer.GetRightWeaponLife() + "/" + pPlayer.GetRightWeaponMaxLife() + "  || WeaponR life: " + pBot.GetRightWeaponLife() + "/" + pBot.GetRightWeaponMaxLife() + "  |");
            Console.WriteLine("|===================-<>-===================|");
            int iPlayerFuel = pPlayer.GetFuel();
            Console.Write("|    Fuel: " + iPlayerFuel + "/100");
            if (iPlayerFuel < 100)
            {
                Console.Write(" ");
            }
            int iBotFuel = pBot.GetFuel();
            Console.Write("   ||    Fuel: " + iBotFuel + "/100");
            if (iBotFuel < 100)
            {
                Console.Write(" ");
            }
            Console.WriteLine("   |");
            Console.WriteLine("|====================--====================|");
        }

        public static int MainMenu(int res = -1)
        {
            int iResult = 0;
            Console.WriteLine("|====================--====================|");
            Console.WriteLine("|                Main  Menu                |");
            Console.WriteLine("|===================-<>-===================|");
            Console.WriteLine("|  1-Attack  ||  2-Repairs  ||  3-Furnace  |");
            Console.WriteLine("|===================-<>-===================|");
            Console.Write("| Tap the number of the desired action : ");
            if(res == -1)
            {
                iResult = Utils.GetInt();
            }
            Console.WriteLine(" |");
            Console.WriteLine("|====================--====================|");
            return iResult;
        }

        public static int TargetMenu(int res = -1)
        {
            int iResult = 0;
            Console.WriteLine("|====================--====================|");
            Console.WriteLine("|              Target  Menu                |");
            Console.WriteLine("|===================-<>-===================|");
            Console.WriteLine("|    1-Left Weapon   ||   2-Right Weapon   |");
            Console.WriteLine("|===================-<>-===================|");
            Console.WriteLine("|       3-Legs       ||     4-Furnace      |");
            Console.WriteLine("|===================-<>-===================|");
            Console.WriteLine("|                    ||       0-Back       |");
            Console.WriteLine("|===================-<>-===================|");
            Console.Write("| Select the target part : ");
            if (res == -1)
            {
                iResult = Utils.GetInt();
            }
            else
            {
                iResult = res;
            }
            Console.WriteLine("               |");
            Console.WriteLine("|====================--====================|");
            return iResult;
        }

        public static int RepairMenu(PILOT.IPILOT pPilot, int res)
        {
            int iResult;
            Console.WriteLine("|====================--====================|");
            Console.WriteLine("|                Kits  Menu                |");
            Console.WriteLine("|===================-<>-===================|");
            List<int> vRepairKitsReserve = pPilot.GetRepairKitsReserve();
            int iLightArmor = vRepairKitsReserve[0];
            Console.Write("| 1-Light armor: x" + iLightArmor);
            if (iLightArmor < 10)
            {
                Console.Write(" ");
            }
            int iHeavyKits = vRepairKitsReserve[1];
            Console.Write(" || 2-Heavy armor: x" + iHeavyKits);
            if (iHeavyKits < 10)
            {
                Console.Write(" ");
            }
            Console.WriteLine(" |");
            Console.WriteLine("|      Armor: 1      ||      Armor: 3      |");
            Console.WriteLine("|===================-<>-===================|");
            int iRepairKits = vRepairKitsReserve[2];
            Console.Write("| 3-Repair kits: x" + iRepairKits);
            if (iRepairKits < 10)
            {
                Console.Write(" ");
            }
            int iFullKits = vRepairKitsReserve[3];
            Console.Write(" ||  4-Full kits: x" + iFullKits);
            if (iFullKits < 10)
            {
                Console.Write(" ");
            }
            Console.WriteLine("  |");
            Console.WriteLine("|   Life Points: 1   ||   Life Points: 3   |");
            Console.WriteLine("|===================-<>-===================|");
            Console.WriteLine("|                    ||       0-Back       |");
            Console.WriteLine("|===================-<>-===================|");
            Console.Write("| Select your kit type : ");
            if (res == -1)
            {
                iResult = Utils.GetInt();
            }
            else
            {
                iResult = res;
            }
            Console.WriteLine("                 |");
            Console.WriteLine("|====================--====================|");
            return iResult;
        }

        public static int FuelMenu(PILOT.IPILOT pPilot, int res = -1)
        {
            int iResult;
            Console.WriteLine("|====================--====================|");
            Console.WriteLine("|                Fuel  Menu                |");
            Console.WriteLine("|===================-<>-===================|");
            List<int> vFuelsReserve = pPilot.GetFuelsReserve();
            int iWood = vFuelsReserve[0];
            Console.Write("|     1-Wood: x" + iWood);
            if (iWood < 10)
            {
                Console.Write(" ");
            }
            int iCharcoal = vFuelsReserve[1];
            Console.Write("    ||   2-Charcoal: x" + iCharcoal);
            if (iCharcoal < 10)
            {
                Console.Write(" ");
            }
            Console.WriteLine("  |");
            Console.WriteLine("|     Energy: 15     ||     Energy: 20     |");
            Console.WriteLine("|===================-<>-===================|");
            int iCoal = vFuelsReserve[2];
            Console.Write("|     3-Coal: x" + iCoal);
            if (iCoal < 10)
            {
                Console.Write(" ");
            }
            int iCompactCoal = vFuelsReserve[3];
            Console.Write("    || 4-Compact Coal: x" + iCompactCoal);
            if (iCompactCoal < 10)
            {
                Console.Write(" ");
            }
            Console.WriteLine("|");
            Console.WriteLine("|     Energy: 25     ||     Energy: 35     |");
            Console.WriteLine("|===================-<>-===================|");
            Console.WriteLine("|                    ||       0-Back       |");
            Console.WriteLine("|===================-<>-===================|");
            Console.Write("| Select your fuel type : ");
            if (res == -1)
            {
                iResult = Utils.GetInt();
            }
            else
            {
                iResult = res;
            }
            Console.WriteLine("                |");
            Console.WriteLine("|====================--====================|");
            return iResult;
        }

        public static int WeaponMenu(Robot pPlayer, int res = -1)
        {
            int iResult;
            Console.WriteLine("|====================--====================|");
            Console.WriteLine("|              Weapon  Menu                |");
            Console.WriteLine("|===================-<>-===================|");
            Console.Write("| 1-Left Weapon type: ");
            switch (pPlayer.GetLeftWeaponType())
            {
                case 1:
                    {
                        Console.WriteLine("Normal               |");
                        break;
                    }
                case 2:
                    {
                        Console.WriteLine("Melee                |");
                        break;
                    }
                case 3:
                    {
                        Console.WriteLine("Projectile           |");
                        int iAmmo = pPlayer.GetLeftWeaponSpecificity();
                        Console.Write("| Remaining ammo: x" + iAmmo);
                        if (iAmmo < 10)
                        {
                            Console.Write(" ");
                        }
                        Console.WriteLine("                      |");
                        break;
                    }
                case 4:
                    {
                        Console.WriteLine("Thermal              |");
                        int iFuel = pPlayer.GetLeftWeaponSpecificity();
                        Console.Write("| Burning fuels: x" + iFuel);
                        if (iFuel < 10)
                        {
                            Console.Write(" ");
                        }
                        Console.WriteLine("                     |");
                        break;
                    }
                default:
                    {
                        Console.WriteLine("Undefined            |");
                        break;
                    }
            }
            Console.WriteLine("|>- - - - - - - - - -<>- - - - - - - - - -<|");
            Console.WriteLine("| Damage: " + pPlayer.GetLeftWeaponDamage() + "                                |");
            int iLeftAccuracy = pPlayer.GetLeftWeaponHitChance();
            Console.Write("| Accuracy: " + iLeftAccuracy + "%");
            if (iLeftAccuracy < 100)
            {
                Console.Write(" ");
            }
            Console.WriteLine("                           |");
            Console.WriteLine("| Fuel Consumption: " + pPlayer.GetLeftWeaponPowerConsumption() + "                     |");
            Console.WriteLine("|===================-<>-===================|");
            Console.Write("| 2-Right Weapon type: ");
            switch (pPlayer.GetRightWeaponType())
            {
                case 1:
                    {
                        Console.WriteLine("Normal              |");
                        break;
                    }
                case 2:
                    {
                        Console.WriteLine("Melee               |");
                        break;
                    }
                case 3:
                    {
                        Console.WriteLine("Projectile          |");
                        int iAmmo = pPlayer.GetRightWeaponSpecificity();
                        Console.Write("| Remaining ammo: x" + iAmmo);
                        if (iAmmo < 10)
                        {
                            Console.Write(" ");
                        }
                        Console.WriteLine("                      |");
                        break;
                    }
                case 4:
                    {
                        Console.WriteLine("Thermal             |");
                        int iFuel = pPlayer.GetRightWeaponSpecificity();
                        Console.Write("| Burning fuels: x" + iFuel);
                        if (iFuel < 10)
                        {
                            Console.Write(" ");
                        }
                        Console.WriteLine("                     |");
                        break;
                    }
                default:
                    {
                        Console.WriteLine("Undefined           |");
                        break;
                    }
            }
            Console.WriteLine("|>- - - - - - - - - -<>- - - - - - - - - -<|");
            Console.WriteLine("| Damage: " + pPlayer.GetRightWeaponDamage() + "                                |");
            int iRightAccuracy = pPlayer.GetRightWeaponHitChance();
            Console.Write("| Accuracy: " + iRightAccuracy + "%");
            if (iRightAccuracy < 100)
            {
                Console.Write(" ");
            }
            Console.WriteLine("                           |");
            Console.WriteLine("| Fuel Consumption: " + pPlayer.GetRightWeaponPowerConsumption() + "                     |");
            Console.WriteLine("|===================-<>-===================|");
            Console.WriteLine("|                    ||       0-Back       |");
            Console.WriteLine("|===================-<>-===================|");
            Console.Write("| Select your weapon : ");
            if(res == -1)
            {
                iResult = Utils.GetInt();
            }
            else
            {
                iResult = res;
            }
            Console.WriteLine("                   |");
            Console.WriteLine("|====================--====================|");
            return iResult;
        }

        public static void RobotRestart()
        {
            Console.WriteLine("|--------------------==--------------------|");
            Console.WriteLine("|      The robot need fuel to restart      |");
            Console.WriteLine("|--------------------==--------------------|");
        }

        public static void WeaponIsUnusable()
        {
            Console.WriteLine("|--------------------==--------------------|");
            Console.WriteLine("|         This weapon  is unusable         |");
            Console.WriteLine("|--------------------==--------------------|");
        }

        public static void AlreadyDestroy()
        {
            Console.WriteLine("|--------------------==--------------------|");
            Console.WriteLine("|       This part is already destroy       |");
            Console.WriteLine("|--------------------==--------------------|");
        }

        public static void MissedFire()
        {
            Console.WriteLine("|--------------------==--------------------|");
            Console.WriteLine("|            It'  missed fire            |");
            Console.WriteLine("|--------------------==--------------------|");
        }

        public static void NoStockKit()
        {
            Console.WriteLine("|--------------------==--------------------|");
            Console.WriteLine("|    No more kits of this type in stock    |");
            Console.WriteLine("|--------------------==--------------------|");
        }

        public static void PerfectlyFine()
        {
            Console.WriteLine("|--------------------==--------------------|");
            Console.WriteLine("|              Perfectly Fine              |");
            Console.WriteLine("|--------------------==--------------------|");
        }

        public static void NoStockFuel()
        {
            Console.WriteLine("|--------------------==--------------------|");
            Console.WriteLine("|    No more fuel of this type in stock    |");
            Console.WriteLine("|--------------------==--------------------|");
        }

        public static void WrongEntry()
        {
            Console.WriteLine("!!!-----------------!==!-----------------!!!");
            Console.WriteLine("|               Wrong  Entry               |");
            Console.WriteLine("!!!-----------------!==!-----------------!!!");
        }
    }
}

File: PJS5_POC_Cpp/Sources/Gui/Gui.cpp
#include "Gui.h"
#include "../Utils.h"
#include "../Pilot/IPilot.h"
#include "../Robot/Robot.h"
#include <iostream>
#include <vector>

namespace GUI
{
	void ShowStatus(ROBOT::ROBOT* pPlayer, ROBOT::ROBOT* pBot)
	{
		std::cout << "|====================--====================|\n";
		std::cout << "|                  Robots                  |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "|>-     Player     -<||>-     Ennemi     -<|\n";
		std::cout << "|===================-<>-===================|\n";
		
		std::cout << "| Furnace armor: " 
			<< pPlayer->GetFurnaceArmor() << "/" << pPlayer->GetFurnaceMaxArmor()
			<< " || Furnace armor: "
			<< pBot->GetFurnaceArmor() << "/" << pBot->GetFurnaceMaxArmor()
			<< " |\n";

		std::cout << "| Furnace life: " 
			<< pPlayer->GetFurnaceLife() << "/" << pPlayer->GetFurnaceMaxLife() 
			<<"  || Furnace life: "
			<< pBot->GetFurnaceLife() << "/" << pBot->GetFurnaceMaxLife()
			<<"  |\n";

		std::cout << "|>- - - - - - - - - -<>- - - - - - - - - -<|\n";

		std::cout << "| Legs armor: "
			<< pPlayer->GetLegsArmor() << "/" << pPlayer->GetLegsMaxArmor()
			<< "    || Legs armor: "
			<< pBot->GetLegsArmor() << "/" << pBot->GetLegsMaxArmor()
			<< "    |\n";

		std::cout << "| Legs remains: "
			<< pPlayer->GetLegsLife() << "/" << pPlayer->GetLegsMaxLife()
			<< "  || Legs remains: "
			<< pBot->GetLegsLife() << "/" << pBot->GetLegsMaxLife()
			<< "  |\n";

		std::cout << "|>- - - - - - - - - -<>- - - - - - - - - -<|\n";
		
		std::cout << "| WeaponL armor: "
			<< pPlayer->GetLeftWeaponArmor() << "/" << pPlayer->GetLeftWeaponMaxArmor()
			<< " || WeaponL armor: "
			<< pBot->GetLeftWeaponArmor() << "/" << pBot->GetLeftWeaponMaxArmor()
			<< " |\n";

		std::cout << "| WeaponL life: "
			<< pPlayer->GetLeftWeaponLife() << "/" << pPlayer->GetLeftWeaponMaxLife()
			<< "  || WeaponL life: "
			<< pBot->GetLeftWeaponLife() << "/" << pBot->GetLeftWeaponMaxLife()
			<< "  |\n";

		std::cout << "|>- - - - - - - - - -<>- - - - - - - - - -<|\n";

		std::cout << "| WeaponR armor: "
			<< pPlayer->GetRightWeaponArmor() << "/" << pPlayer->GetRightWeaponMaxArmor()
			<< " || WeaponR armor: "
			<< pBot->GetRightWeaponArmor() << "/" << pBot->GetRightWeaponMaxArmor()
			<< " |\n";

		std::cout << "| WeaponR life: "
			<< pPlayer->GetRightWeaponLife() << "/" << pPlayer->GetRightWeaponMaxLife()
			<< "  || WeaponR life: "
			<< pBot->GetRightWeaponLife() << "/" << pBot->GetRightWeaponMaxLife()
			<< "  |\n";

		std::cout << "|===================-<>-===================|\n";

		const int iPlayerFuel = pPlayer->GetFuel();
		std::cout << "|    Fuel: " << iPlayerFuel << "/100";
		if (iPlayerFuel < 100)
		{
			std::cout << " ";
		}
		
		const int iBotFuel = pBot->GetFuel();
		std::cout << "   ||    Fuel: " << iBotFuel << "/100";
		if (iBotFuel < 100)
		{
			std::cout << " ";
		}
		std::cout <<"   |\n";

		std::cout << "|====================--====================|\n";
	}

	int MainMenu()
	{
		int iResult;
		std::cout << "|====================--====================|\n";
		std::cout << "|                Main  Menu                |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "|  1-Attack  ||  2-Repairs  ||  3-Furnace  |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "| Tap the number of the desired action : ";
		iResult = UTILS::GetInt();
		std::cout << " |\n";
		std::cout << "|====================--====================|\n";

		return iResult;
	}

	int TargetMenu()
	{
		int iResult;

		std::cout << "|====================--====================|\n";
		std::cout << "|              Target  Menu                |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "|    1-Left Weapon   ||   2-Right Weapon   |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "|       3-Legs       ||     4-Furnace      |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "|                    ||       0-Back       |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "| Select the target part : ";
		iResult = UTILS::GetInt();
		std::cout << "               |\n";
		std::cout << "|====================--====================|\n";

		return iResult;
	}

	int RepairMenu(PILOT::IPILOT* pPilot) 
	{
		int iResult;

		std::cout << "|====================--====================|\n";
		std::cout << "|                Kits  Menu                |\n";
		std::cout << "|===================-<>-===================|\n";
		
		const std::vector<int> vRepairKitsReserve = pPilot->GetRepairKitsReserve();

		const int iLightArmor = vRepairKitsReserve.at(0);
		std::cout << "| 1-Light armor: x" << iLightArmor;
		if (iLightArmor < 10)
		{
			std::cout << " ";
		}

		const int iHeavyKits = vRepairKitsReserve.at(1);
		std::cout << " || 2-Heavy armor: x" << iHeavyKits;
		if (iHeavyKits < 10)
		{
			std::cout << " ";
		}
		std::cout << " |\n";

		std::cout << "|      Armor: 1      ||      Armor: 3      |\n";
		std::cout << "|===================-<>-===================|\n";

		const int iRepairKits = vRepairKitsReserve.at(2);
		std::cout << "| 3-Repair kits: x" << iRepairKits;
		if (iRepairKits < 10)
		{
			std::cout << " ";
		}

		const int iFullKits = vRepairKitsReserve.at(3);
		std::cout << " ||  4-Full kits: x" << iFullKits;
		if (iFullKits < 10)
		{
			std::cout << " ";
		}
		std::cout << "  |\n";

		std::cout << "|   Life Points: 1   ||   Life Points: 3   |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "|                    ||       0-Back       |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "| Select your kit type : ";
		iResult = UTILS::GetInt();
		std::cout << "                 |\n";
		std::cout << "|====================--====================|\n";

		return iResult;
	}

	int FuelMenu(PILOT::IPILOT* pPilot)
	{
		int iResult;

		std::cout << "|====================--====================|\n";
		std::cout << "|                Fuel  Menu                |\n";
		std::cout << "|===================-<>-===================|\n";

		const std::vector<int> vFuelsReserve = pPilot->GetFuelsReserve();

		const int iWood = vFuelsReserve.at(0);
		std::cout << "|     1-Wood: x" << iWood;
		if (iWood < 10)
		{
			std::cout << " ";
		}

		const int iCharcoal = vFuelsReserve.at(1);
		std::cout << "    ||   2-Charcoal: x" << iCharcoal;
		if (iCharcoal < 10)
		{
			std::cout << " ";
		}
		std::cout << "  |\n";
		
		std::cout << "|     Energy: 15     ||     Energy: 20     |\n";
		std::cout << "|===================-<>-===================|\n";
		
		const int iCoal = vFuelsReserve.at(2);
		std::cout << "|     3-Coal: x" << iCoal;
		if (iCoal < 10)
		{
			std::cout << " ";
		}

		const int iCompactCoal = vFuelsReserve.at(3);
		std::cout << "    || 4-Compact Coal: x" << iCompactCoal;
		if (iCompactCoal < 10)
		{
			std::cout << " ";
		}
		std::cout << "|\n";

		std::cout << "|     Energy: 25     ||     Energy: 35     |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "|                    ||       0-Back       |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "| Select your fuel type : ";
		iResult = UTILS::GetInt();
		std::cout << "                |\n";
		std::cout << "|====================--====================|\n";

		return iResult;
	}

	int WeaponMenu(ROBOT::ROBOT* pPlayer)
	{
		int iResult;

		std::cout << "|====================--====================|\n";
		std::cout << "|              Weapon  Menu                |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "| 1-Left Weapon type: ";
		switch (pPlayer->GetLeftWeaponType())
		{
		case 1:
		{
			std::cout << "Normal               |\n";
			break;
		}
		case 2:
		{
			std::cout << "Melee                |\n";
			break;
		}
		case 3:
		{
			std::cout << "Projectile           |\n";
			const int iAmmo = pPlayer->GetLeftWeaponSpecificity();
			std::cout << "| Remaining ammo: x" << iAmmo;
			if (iAmmo < 10)
			{
				std::cout << " ";
			}
			std::cout << "                      |\n";
			break;
		}
		case 4:
		{
			std::cout << "Thermal              |\n";
			const int iFuel = pPlayer->GetLeftWeaponSpecificity();
			std::cout << "| Burning fuels: x" << iFuel;
			if (iFuel < 10)
			{
				std::cout << " ";
			}
			std::cout << "                     |\n";
			break;
		}
		default :
		{
			std::cout << "Undefined            |\n";
			break;
		}
		}
		std::cout << "|>- - - - - - - - - -<>- - - - - - - - - -<|\n";
		std::cout << "| Damage: " << pPlayer->GetLeftWeaponDamage() << "                                |\n";

		const int iLeftAccuracy = pPlayer->GetLeftWeaponHitChance();
		std::cout << "| Accuracy: " << iLeftAccuracy << "%";
		if (iLeftAccuracy < 100)
		{
			std::cout << " ";
		}
		std::cout << "                           |\n";
		std::cout << "| Fuel Consumption: " << pPlayer->GetLeftWeaponPowerConsumption() << "                     |\n";


		std::cout << "|===================-<>-===================|\n";
		
		std::cout << "| 2-Right Weapon type: ";
		switch (pPlayer->GetRightWeaponType())
		{
		case 1:
		{
			std::cout << "Normal              |\n";
			break;
		}
		case 2:
		{
			std::cout << "Melee               |\n";
			break;
		}
		case 3:
		{
			std::cout << "Projectile          |\n";
			const int iAmmo = pPlayer->GetRightWeaponSpecificity();
			std::cout << "| Remaining ammo: x" << iAmmo;
			if (iAmmo < 10)
			{
				std::cout << " ";
			}
			std::cout << "                      |\n";
			break;
		}
		case 4:
		{
			std::cout << "Thermal             |\n";
			const int iFuel = pPlayer->GetRightWeaponSpecificity();
			std::cout << "| Burning fuels: x" << iFuel;
			if (iFuel < 10)
			{
				std::cout << " ";
			}
			std::cout << "                     |\n";
			break;
		}
		default:
		{
			std::cout << "Undefined           |\n";
			break;
		}
		}
		std::cout << "|>- - - - - - - - - -<>- - - - - - - - - -<|\n";
		std::cout << "| Damage: " << pPlayer->GetRightWeaponDamage() << "                                |\n";

		const int iRightAccuracy = pPlayer->GetRightWeaponHitChance();
		std::cout << "| Accuracy: " << iRightAccuracy << "%";
		if (iRightAccuracy < 100)
		{
			std::cout << " ";
		}
		std::cout << "                           |\n";
		std::cout << "| Fuel Consumption: " << pPlayer->GetRightWeaponPowerConsumption() << "                     |\n";

		std::cout << "|===================-<>-===================|\n";
		std::cout << "|                    ||       0-Back       |\n";
		std::cout << "|===================-<>-===================|\n";
		std::cout << "| Select your weapon : ";
		iResult = UTILS::GetInt();
		std::cout << "                   |\n";
		std::cout << "|====================--====================|\n";

		return iResult;
	}

	void RobotRestart()
	{
		std::cout << "|--------------------==--------------------|\n";
		std::cout << "|      The robot need fuel to restart      |\n";
		std::cout << "|--------------------==--------------------|\n";
	}

	void WeaponIsUnusable()
	{
		std::cout << "|--------------------==--------------------|\n";
		std::cout << "|         This weapon  is unusable         |\n";
		std::cout << "|--------------------==--------------------|\n";
	}

	void AlreadyDestroy()
	{
		std::cout << "|--------------------==--------------------|\n";
		std::cout << "|       This part is already destroy       |\n";
		std::cout << "|--------------------==--------------------|\n";
	}

	void MissedFire()
	{
		std::cout << "|--------------------==--------------------|\n";
		std::cout << "|            It's a missed fire            |\n";
		std::cout << "|--------------------==--------------------|\n";
	}

	void NoStockKit()
	{
		std::cout << "|--------------------==--------------------|\n";
		std::cout << "|    No more kits of this type in stock    |\n";
		std::cout << "|--------------------==--------------------|\n";
	}

	void PerfectlyFine()
	{
		std::cout << "|--------------------==--------------------|\n";
		std::cout << "|              Perfectly Fine              |\n";
		std::cout << "|--------------------==--------------------|\n";
	}

	void NoStockFuel()
	{
		std::cout << "|--------------------==--------------------|\n";
		std::cout << "|    No more fuel of this type in stock    |\n";
		std::cout << "|--------------------==--------------------|\n";
	}

	void WrongEntry()
	{
		std::cout << "!!!-----------------!==!-----------------!!!\n";
		std::cout << "|               Wrong  Entry               |\n";
		std::cout << "!!!-----------------!==!-----------------!!!\n";
	}
}